packages/shopping/src/__tests__/acceptance/user.controller.acceptance.ts

```diff
@@ -100,106 +96,155 @@ describe('UserController', () => {
+      expect(res.body.error.message).to.equal(
+        'The credentials are not correct.',
+      );
     });
 
-    it('/me returns the current user', async () => {
-      const newUser = await userRepo.create(user);
-      const token = await jwtAuthService.getAccessTokenForUser({
-        email: newUser.email,
-        password: plainPassword,
-      });
+    it('users/me returns the current user profile when a valid JWT token is provided', async () => {
+      let res = await client
+        .post('/users')
+        .send(user)
+        .expect(200);
 
-      newUser.id = newUser.id.toString();
-      const me = _.pick(toJSON(newUser), ['id', 'email']);
+      const createdUserId = res.body.id;
+      res = await client
+        .post('/users/login')
+        .send({email: user.email, password: user.password})
+        .expect(200);
 
-      await client
+      const token = res.body.token;
+
+      res = await client
         .get('/users/me')
         .set('Authorization', 'Bearer ' + token)
-        .expect(200, me);
+        .expect(200);
+
+      const userProfile = res.body;
+      expect(userProfile.id).to.equal(createdUserId);
+      expect(userProfile.name).to.equal(`${user.firstname} ${user.surname}`);
     });
 
-    it('/me returns 401 when the token is not provided', async () => {
-      const newUser = await userRepo.create(user);
-      await jwtAuthService.getAccessTokenForUser({
-        email: newUser.email,
-        password: plainPassword,
-      });
+    it('users/me returns an error when a JWT token is not provided', async () => {
+      const res = await client.get('/users/me').expect(401);
+
+      expect(res.body.error.message).to.equal(
+        'Authorization header not found.',
+      );
+    });
 
-      await client.get('/users/me').expect(401);
+    it('users/me returns an error when an invalid JWT token is provided', async () => {
+      const res = await client
+        .get('/users/me')
+        .set('Authorization', 'Bearer ' + 'xxx.yyy.zzz')
+        .expect(401);
+
+      expect(res.body.error.message).to.equal(
+        'Error verifying token : invalid token',
+      );
+    });
+
+    it(`users/me returns an error when 'Bearer ' is not found in Authorization header`, async () => {
+      const res = await client
+        .get('/users/me')
+        .set('Authorization', 'NotB3@r3r ' + 'xxx.yyy.zzz')
+        .expect(401);
+
+      expect(res.body.error.message).to.equal(
+        "Authorization header is not of type 'Bearer'.",
+      );
+    });
+
+    it('users/me returns an error when an expired JWT token is provided', async () => {
+      const res = await client
+        .get('/users/me')
+        .set('Authorization', 'Bearer ' + getExpiredToken())
+        .expect(401);
+
+      expect(res.body.error.message).to.equal(
+        'Error verifying token : jwt expired',
+      );
+    });
+  });
+
+  describe('user product recommendation (service) api', () => {
+    // tslint:disable-next-line:no-any
```

Comment:
Is this line needed?

Reply:
@nabdelgadir , good catch. This is an old comment. This particular one is not needed, but there is another elsewhere that is needed. Good catch.